html/php-pre-challenge3/index.php

```diff
@@ -1,6 +1,70 @@
 <?php
+
+// URLで入力されるtarget値$limitに代入している
 $limit = $_GET['target'];
 
+// db接続情報
 $dsn = 'mysql:dbname=test;host=mysql';
 $dbuser = 'test';
 $dbpassword = 'test';
+
+// dbへ接続と接続できなかった際の表示を設定
+try {
+  // pdoクラスのインスタンス化
+  $pdo = new PDO($dsn, $dbuser, $dbpassword);
+} catch (PDOException $e) {
+  echo 'DBに接続できていません' . $e->getMessage();
+}
+
+// prechallenege内の数字をaryListに格納
+$sql = "SELECT value FROM prechallenge3";
+$sth = $pdo->query($sql);
+$aryList = $sth->fetchAll(PDO::FETCH_COLUMN);
+
+// 選ぶ元となる全体を配列で与え、抜き取る数を与えると全ての組み合わせを配列で返す関数
+function combination($whole, $pickupNumber)
+{
+  $wholeNumber = count($whole);
+  if ($wholeNumber < $pickupNumber) {
+    return;
+  } elseif ($pickupNumber == 1) {
+    for ($i = 0; $i < $wholeNumber; $i++) {
+      $arrs[$i] = array($whole[$i]);
+    }
+  } elseif ($pickupNumber > 1) {
+    $j = 0;
+    for ($i = 0; $i < $wholeNumber - $pickupNumber + 1; $i++) {
+      $cuts = combination(array_slice($whole, $i + 1), $pickupNumber - 1);
+      foreach ($cuts as $cut) {
+        array_unshift($cut, $whole[$i]);
+        $arrs[$j] = $cut;
+        $j++;
+      }
+    }
+  }
+  return $arrs;
+}
+
+// 合計数$numと組み合わせ処理で得た数が同一かどうか調べ、同一であれば答えの配列に入れる関数
+function find_combinations($lis, $num)
+{
+  $answer_list = [];
+  for ($i = 1; $i < count($lis) + 1; $i++) {
```

Comment:
```suggestion
  $answer_list = [];
  $length = count($lis);
  for ($i = 1; $i < $length + 1; $i++) {
```
今だとcount関数をfor文でループ事に実行してしまうので、計算結果を変数に入れることでcount関数の実行を1回で済むようにしましょう。

Reply:
9453a17 こちらのコミットで実装しました。